lms/services/youtube.py

```diff
@@ -58,11 +58,21 @@ def video_info(self, video_id: str) -> dict:
             raise VideoNotFound(video_id) from err
 
         snippet = item["snippet"]
+        content_details = item["contentDetails"]
+
+        # ytRating field docs: https://developers.google.com/youtube/v3/docs/videos#contentDetails.contentRating.ytRating
+        is_restricted = (
+            False
+            if "ytRating" not in content_details["contentRating"]
+            else content_details["contentRating"]["ytRating"] == "ytAgeRestricted"
+        )
+
         return {
             "image": snippet["thumbnails"]["medium"]["url"],
             "title": snippet["title"],
             "channel": snippet["channelTitle"],
-            "duration": item["contentDetails"]["duration"],  # ISO duration
+            "duration": content_details["duration"],  # ISO duration
+            "is_restricted": is_restricted,
```

Comment:
Yes, I intentionally used a broader name, so that we can include other reasons for the video to be restricted.

@robertknight has identified a couple more, for example non-embeddable videos, or videos not accessible from certain regions.

That makes me think that `is_restricted` is actually not enough, and we should probably have some way to identify the reason for a video to be restricted.

Possible options I can think of:

* Add one more `restriction_reason` field where we set the reason of the restriction (should it be nullable? optional? What do you usually do in these cases).
* Wrap `is_restricted` and any other field related with restrictions into a wrapping, let's say, ''restrictions` property.
  ```json
  {
    "title": "Hypothesis and Atlassian New Partnership Announced at the Team23 Conference",
    // [...]
    "restrictions": {
      // Whatever we feel like puttin ghere
    }
  }
  ```
* Replace everything with a list of restriction reasons `"restriction_resons": ["age", "non_embeddable"]` (perhaps a video is both age-restricted and non-embeddable), and if the list is empty, we know it's not restricted at all